### DISCUSSION TITLE
`disk full error` in the admin model download setting with free disk space indicator.

### DISCUSSION THREAD

**Discussion by hemangjoshi37a** [open-webui/open-webui#4151] [General]
please add a error or warning or free disk space showing right beside the model download dialogue box in the admin setting so we can know if we can download certainly big model or not . this is for the people who are running ollama in the cloud . thank you.

**Comment by justinh-rahb** [comment]
There's no way for WebUI to know how much disk space is available for Ollama, the information is not provided in the API.

**Reply by hemangjoshi37a** [reply]
we can use some command like `lsblk` running in the bash and get the result and show that in the UI.

**Reply by justinh-rahb** [reply]
Now you need an agent that installs on the Ollama host to report this back to WebUI... this is a lot of moving parts. Can't see it happening.

**Comment by stsazama** [comment]
Would it be reasonable to add an optional setting that limits the total size of models that WebUI is willing to download? That way the WebUI doesn't need to check how much disk space is available for Ollama, only keep track of how much it has contributed itself. The server manager can decide what to set that limit at.
It would be very useful to give users some indication of how much space they have for models before they hit an error.
